File: Edit-Profile-Service/src/main/kotlin/com/kinok0/editprofileservice/entity/RefreshTokenEntity.java
package com.kinok0.editprofileservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "refresh_tokens")
@Getter
@Setter
@NoArgsConstructor
public class RefreshTokenEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"user\"", referencedColumnName = "id", nullable = false)
    private UserEntity user;

    @Column(name = "token", nullable = false, unique = true)
    private String token;

    public RefreshTokenEntity(UserEntity user, String token) {
        this.user = user;
        this.token = token;
    }
}

File: Profile-Service/src/main/kotlin/com/kinok0/profileservice/entity/RefreshTokenEntity.kt
package com.kinok0.profileservice.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "refresh_tokens")
data class RefreshTokenEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"user\"", referencedColumnName = "id", nullable = false)
    var user: UserEntity,

    @Column(name = "token", nullable = false, unique = true)
    var token: String
)
